pkg/minikube/cluster/cluster_linux.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"fmt"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/docker/machine/libmachine/drivers"
+	"github.com/minishift/minishift/pkg/minikube/constants"
+)
+
+type kvmDriver struct {
+	*drivers.BaseDriver
+
+	Memory         int
+	DiskSize       int
+	CPU            int
+	Network        string
+	PrivateNetwork string
+	ISO            string
+	Boot2DockerURL string
+	DiskPath       string
+	CacheMode      string
+	IOMode         string
+}
+
+func createKVMHost(config MachineConfig) *kvmDriver {
+	return &kvmDriver{
+		BaseDriver: &drivers.BaseDriver{
+			MachineName: constants.MachineName,
+			StorePath:   constants.Minipath,
+			SSHUser:     "docker",
+		},
+		Memory:         config.Memory,
+		CPU:            config.CPUs,
+		PrivateNetwork: "docker-machines",
+		Network:        "default",
+		Boot2DockerURL: config.GetISOFileURI(),
+		DiskSize:       config.DiskSize,
+		DiskPath:       filepath.Join(constants.Minipath, "machines", constants.MachineName, fmt.Sprintf("%s.img", constants.MachineName)),
+		ISO:            filepath.Join(constants.Minipath, "machines", constants.MachineName, "boot2docker.iso"),
+		CacheMode:      "default",
+		IOMode:         "threads",
+	}
+}
+
+func createKVM2Host(config MachineConfig) *kvmDriver {
+	return &kvmDriver{
+		BaseDriver: &drivers.BaseDriver{
+			MachineName: constants.MachineName,
+			StorePath:   constants.Minipath,
+			SSHUser:     "docker",
+		},
+		Memory:         config.Memory,
+		CPU:            config.CPUs,
+		PrivateNetwork: "minikube-net",
+		Network:        "default",
+		Boot2DockerURL: config.GetISOFileURI(),
+		DiskSize:       config.DiskSize,
+		DiskPath:       filepath.Join(constants.Minipath, "machines", constants.MachineName, fmt.Sprintf("%s.rawdisk", constants.MachineName)),
+		ISO:            filepath.Join(constants.Minipath, "machines", constants.MachineName, "boot2docker.iso"),
+		CacheMode:      "default",
+		IOMode:         "threads",
+	}
+}
+
+func detectVBoxManageCmd() string {
+	cmd := "VBoxManage"
+	if path, err := exec.LookPath(cmd); err == nil {
+		return path
+	}
+	return cmd
+}
```

Comment:
This something @anjannath added for #1822. Also we should remove `detectVBoxManageCmd` functions as these are not used anywhere.

Reply:
We (@praveenkumar and I) actually discussed these methods with the Minikube team, and we would like to use these methods instead for doing the preflight checks. As the implementation is driver specific, only the driver should handle their own verification. This will also eventually lead to a possible install of drivers #1778, which is handled by the driver plugin.